File: tests/fixtures/sample_go/repository.go
package sample

// UserRepository stores users keyed by name.
type UserRepository struct {
	users map[string]*User
}

// NewUserRepository builds an empty repository.
func NewUserRepository() *UserRepository {
	return &UserRepository{users: make(map[string]*User)}
}

func (r *UserRepository) Add(u *User) {
	r.users[u.Name] = u
}

func (r *UserRepository) Get(name string) *User {
	return r.users[name]
}

// Count returns the number of stored users.
func (r *UserRepository) Count() int {
	// BUG: off-by-one; should return len(r.users).
	return len(r.users) + 1
}

File: tests/fixtures/sample_ts/repository.ts
import type { User } from "./models";

export class UserRepository {
  private users: Map<string, User> = new Map();

  add(user: User): void {
    this.users.set(user.name, user);
  }

  get(name: string): User | undefined {
    return this.users.get(name);
  }

  count(): number {
    // BUG: off-by-one; should return this.users.size.
    return this.users.size + 1;
  }
}
